src/components/Navbar.js

```diff
@@ -135,5 +136,15 @@ const Navbar = ({ toggleDarkMode, mode }) => {
     </AppBar>
   );
 };
-
+Navbar.propTypes = {
+  toggleDarkMode: PropTypes.func.isRequired,
+  mode: PropTypes.string.isRequired,
+  handleClearCart: PropTypes.func.isRequired,
+  isLoggedIn: PropTypes.bool,
+  statusOptions: PropTypes.oneOf(['active', 'inactive']),
+  userDetails: PropTypes.shape({
+    username: PropTypes.string,
+    email: PropTypes.string,
+  }),
+};
```

Comment:
`propTypes` is only used for documenting the `props` of a component. For function components, that means, the arguments in the function definition `const Navbar ({ toggleDarkMode, mode}) => ...` . So we can remove anything that is not a prop; otherwise, right now we get warnings like this in the browser console:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/f687bc01-d56a-46f2-a979-fee7123136c6)

Unfortunately, there's not really a straightforward way of documenting internal variables of a component, or state variables, apart from using JSDoc (similar to how we document our API specs in the backend) or Typescript. 

Additionally, we can make `mode` even more specific by using the `propTypes.oneOf` function which allows us to tell the prop checker that this value can only be a specific set of values (in this case, we expect mode to be either `dark` or `light` and would want to get warned if it somehow was being set to anything else)

```suggestion
Navbar.propTypes = {
  toggleDarkMode: PropTypes.func.isRequired,
  mode: PropTypes.oneOf(['light', 'dark']).isRequired,
};
```

And example of how this works; I set mine to `oneOf(['dark', 'medium'])` and it showed the following warning in the browser console:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/c1f3bde6-1492-4471-bbdd-593586bd4ee3)